analyzer_test.go

```diff
@@ -17,40 +17,24 @@ const testValue = "hello_world"
 func TestAnalyzer(t *testing.T) {
 	t.Parallel()
 
-	testCases := []struct {
-		dir string
-	}{
-		{
-			dir: "alias",
-		},
-
-		{
-			dir: "common",
-		},
-
-		{
-			dir: "embedded",
-		},
-
-		{
-			dir: "global",
-		},
-
-		{
-			dir: "struct",
-		},
+	testCases := [...]string{
```

Comment:
Hi

Can you explain me the [...]string syntax ?

I faced it only a few times in the past years, so I'm curious to know why you used it?

Does this syntax have a name ?

Reply:
Hi, @ccoVeille!

### What is it:
`[...]ElemtType`:  is array of ElemntType, length equal to the maximum element index plus one.
 `[]EelmentTyp`: is slice.
 
```go
[]string{} // literal for slices
```

```go
[...]string{"1"} // same as [1]string{"1"}, literal for arrays
```
https://go.dev/play/p/tCthh4xSCeH

### Why I used this:
It is micro optimization, but it is free (the code is no more complicated). 
Golang knows the size of array more likely to put it on the stack, not on the heap.

Asm: https://go.godbolt.org/z/5T775edT9


### Does this syntax have a name:

No, It doesn't have name, go spec says only `The notation ... specifies`

https://go.dev/ref/spec#LiteralType

<img width="1188" alt="image" src="https://github.com/lasiar/canonicalheader/assets/31652715/773edd50-dd56-4cea-b292-e4a6785de3bd">

----
Best regards,
Roman

Reply:
Thanks for your detailed reply

I assumed it was something about heap and stack.

I like learning things even if I'm experienced, the more you know, the less often you learn, the more valuable it becomes.

Christophe